### PR TITLE
postgresqlPackages.anonymizer: init at 1.3.1; add me & osnyx to flyingcircus team

### PR DESCRIPTION
**Pull request by Ma27**
## Description of changes

* Adds me and @osnyx to the flyingcircus team (we discussed this directly, but please approve here as well for the record :)). Also needs an approval from one of the other team members (cc @dpausp @ctheune @frlan @leona-ya )
* Adds the postgresql extension [pg_anonymizer](https://gitlab.com/dalibo/postgresql_anonymizer).

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by leona-ya — APPROVED** [review]

**Review by osnyx — APPROVED** [review]
I approve the part where I am added as a maintainer; I have not (yet) looked at the postgres anonymizer part though.

**Review by osnyx — APPROVED** [review]
Now also approving the full packaging and testing code.

**Comment by Ma27** [comment]
Updated to 1.3.1, the latest version of the package being available.

**Review by leona-ya — CHANGES_REQUESTED** [review]
The compilation of the `postgresqlJitPackages.anonymizer` package fails for me (for all psql versions) with:

```
install -d /nix/store/cdsqi903x9lik27rxh9yxsm5zczwvxyw-postgresql_anonymizer-1.3.1/bin
install -m 0755 bin/pg_dump_anon.sh /nix/store/cdsqi903x9lik27rxh9yxsm5zczwvxyw-postgresql_anonymizer-1.3.1/bin
/nix/store/mb488rr560vq1xnl10hinnyfflcrd51n-coreutils-9.4/bin/install -c -m 644 .//anon.control '/nix/store/cdsqi903x9lik27rxh9yxsm5zczwvxyw-postgresql_anonymizer-1.3.1/share/postgresql/extension/'
/nix/store/mb488rr560vq1xnl10hinnyfflcrd51n-coreutils-9.4/bin/install -c -m 644 .//anon/*  '/nix/store/cdsqi903x9lik27rxh9yxsm5zczwvxyw-postgresql_anonymizer-1.3.1/share/postgresql/extension/anon/'
/nix/store/mb488rr560vq1xnl10hinnyfflcrd51n-coreutils-9.4/bin/install -c -m 755  anon.so '/nix/store/cdsqi903x9lik27rxh9yxsm5zczwvxyw-postgresql_anonymizer-1.3.1/lib/'
/nix/store/mb488rr560vq1xnl10hinnyfflcrd51n-coreutils-9.4/bin/mkdir -p '/nix/store/cdsqi903x9lik27rxh9yxsm5zczwvxyw-postgresql_anonymizer-1.3.1/lib/bitcode/anon'
/nix/store/mb488rr560vq1xnl10hinnyfflcrd51n-coreutils-9.4/bin/mkdir -p '/nix/store/cdsqi903x9lik27rxh9yxsm5zczwvxyw-postgresql_anonymizer-1.3.1/lib/bitcode'/anon/
/nix/store/mb488rr560vq1xnl10hinnyfflcrd51n-coreutils-9.4/bin/install -c -m 644 anon.bc '/nix/store/cdsqi903x9lik27rxh9yxsm5zczwvxyw-postgresql_anonymizer-1.3.1/lib/bitcode'/anon/./
cd '/nix/store/cdsqi903x9lik27rxh9yxsm5zczwvxyw-postgresql_anonymizer-1.3.1/lib/bitcode' && llvm-lto -thinlto -thinlto-action=thinlink -o anon.index.bc anon/anon.bc
/nix/store/4vzal97iq3dmrgycj8r0gflrh51p8w1s-bash-5.2p26/bin/bash: line 1: llvm-lto: command not found
make: *** [/nix/store/8hkjv5idbsal1rf5b8hmc43rx2k0gzqs-postgresql-13.14/lib/pgxs/src/makefiles/pgxs.mk:235: install] Error 127
```

I don't know whether pganon supports JIT at all, but we should either fix that or mark as broken / remove pganon from the jit packages.

Without JIT it lgtm.

**Comment by Ma27** [comment]
@leona-ya seems like adding `llvm` for that to `nativeBuildInputs` is sufficient, seems to be working fine as well.

For the record, stdenv doesn't need extra configuration, when using JIT, there's already an LLVM stdenv passed into the extensions.

**Review by leona-ya — APPROVED** [review]

**Comment by Ma27** [comment]
Pushed another commit which adds the `pg_dump_anon` program. This can be used on a remote database, i.e. without the extension part, hence the new package. I decided against using multiple outputs here since it's a Go package and the Go builder doesn't seem composable, so I'd have to split up the build processes anyways.

Also extended the test to make sure the program does what it advertises, providing database dumps, but with data being anonymized.

cc @leona-ya @osnyx

File: nixos/tests/pg_anonymizer.nix
Comment (osnyx) [inline nixos/tests/pg_anonymizer.nix:52]: It's a bit wild that pg_dump and pg_dump_anon appear to use different formats, but I guess that's not your fault.

**Review by osnyx — APPROVED** [review]

File: pkgs/servers/sql/postgresql/ext/anonymizer.nix
Comment (marsam) [inline pkgs/servers/sql/postgresql/ext/anonymizer.nix:30]: nitpick
```suggestion
    description = "Extension to mask or replace personally identifiable information (PII) or commercially sensitive data from a PostgreSQL database";
```

**Review by marsam — APPROVED** [review]
LGTM

**Review by leona-ya — APPROVED** [review]

**Pull request merged**
